### PR TITLE
feat: add op-deployer inspect command

### PR DESCRIPTION
**Pull request by mds1**
Part of https://github.com/ethereum-optimism/platforms-team/issues/284

Adds `l1` subcommand to `op-deployer inspect` so you can now run the below command.

```
$ op-deployer inspect l1 <l2-chain-id>
{
  "superchainDeployment": {
    "proxyAdminAddress": "0xdee57160aafcf04c34c887b5962d0a69676d3c8b",
    "superchainConfigProxyAddress": "0xc2be75506d5724086deb7245bd260cc9753911be",
    "superchainConfigImplAddress": "0x0000000000000000000000000000000000000000",
    "protocolVersionsProxyAddress": "0x79add5713b383daa0a138d3c4780c7a1804a8090",
    "protocolVersionsImplAddress": "0x0000000000000000000000000000000000000000"
  },
  "opChainDeployment": {
    "proxyAdminAddress": "0x59788b9bde63ff99582e565d749234d2accd74db",
    "addressManagerAddress": "0x22308bdbbca059e11d2f793cf42d4d56acacfb25",
    "l1ERC721BridgeProxyAddress": "0x7c1784531c228791c7bbb0274cab13eae3ecd8c6",
    "systemConfigProxyAddress": "0x6942c82bf49388083d47ec485925ba6a6b059204",
    "optimismMintableERC20FactoryProxyAddress": "0x69f5003cb43a0781ebb0c6b26d23b2346ddcbffc",
    "l1StandardBridgeProxyAddress": "0x5915c05a119c460051d8b06516e16ea256670500",
    "l1CrossDomainMessengerProxyAddress": "0xfd524201c927b1a4d3851f1f6544ac80fc5f6ef4",
    "optimismPortalProxyAddress": "0xf56ee26e2abf87dedd99913ad8ed6c0e58a30b49",
    "disputeGameFactoryProxyAddress": "0xb8aaf32a25a31d1b9f777b42990dd3bf997e1d31",
    "anchorStateRegistryProxyAddress": "0xcbdbd22f5b7697f77d85d199a0ee960d2fae8cf1",
    "anchorStateRegistryImplAddress": "0xec3b8f56ffda794b71076fda1f35971abea19d1f",
    "permissionedDisputeGameAddress": "0xe2e9aa6b7410afade1a94de341ebe9c93a8f1f33",
    "delayedWETHPermissionedGameProxyAddress": "0xabe89f9170148ec291e5a84d7586096b03451952"
  },
  "implementationsDeployment": {
    "opcmProxyAddress": "0x0f29118caed0f72873701bcc079398c594b6f8e4",
    "delayedWETHImplAddress": "0x0000000000000000000000000000000000000000",
    "optimismPortalImplAddress": "0x0000000000000000000000000000000000000000",
    "preimageOracleSingletonAddress": "0x0000000000000000000000000000000000000000",
    "mipsSingletonAddress": "0x0000000000000000000000000000000000000000",
    "systemConfigImplAddress": "0x0000000000000000000000000000000000000000",
    "l1CrossDomainMessengerImplAddress": "0x0000000000000000000000000000000000000000",
    "l1ERC721BridgeImplAddress": "0x0000000000000000000000000000000000000000",
    "l1StandardBridgeImplAddress": "0x0000000000000000000000000000000000000000",
    "optimismMintableERC20FactoryImplAddress": "0x0000000000000000000000000000000000000000",
    "disputeGameFactoryImplAddress": "0x0000000000000000000000000000000000000000"
  }
}
```

You'll notice a lot of addresses are the zero address. They are also the zero address in `state.json`, and fixing this is tracked under the "In `state.json`, some implementations and other addresses are the zero address" bullet of the linked issue

### CODE REVIEW

File: op-chain-ops/deployer/inspect/l1.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	L1ERC721BridgeProxyAddress               common.Address `json:"l1ERC721BridgeProxyAddress"`
+	SystemConfigProxyAddress                 common.Address `json:"systemConfigProxyAddress"`
+	OptimismMintableERC20FactoryProxyAddress common.Address `json:"optimismMintableERC20FactoryProxyAddress"`
+	L1StandardBridgeProxyAddress             common.Address `json:"l1StandardBridgeProxyAddress"`
+	L1CrossDomainMessengerProxyAddress       common.Address `json:"l1CrossDomainMessengerProxyAddress"`
+	OptimismPortalProxyAddress               common.Address `json:"optimismPortalProxyAddress"`
+	DisputeGameFactoryProxyAddress           common.Address `json:"disputeGameFactoryProxyAddress"`
+	AnchorStateRegistryProxyAddress          common.Address `json:"anchorStateRegistryProxyAddress"`
+	AnchorStateRegistryImplAddress           common.Address `json:"anchorStateRegistryImplAddress"`
+	// FaultDisputeGameAddress                  common.Address `json:"faultDisputeGameAddress"`
```
Comment (mds1) [inline op-chain-ops/deployer/inspect/l1.go:39]: We don't currently deploy the permissionless game contracts, so left these out for now and kept the comments as placeholders

File: op-chain-ops/deployer/inspect/flags.go [hunk 2]
```diff
@@ -28,6 +28,14 @@ var Flags = []cli.Flag{
 }
 
 var Commands = []*cli.Command{
+	{
+		Name:      "l1",
+		Usage:     "outputs all L1 contract addresses for an L2 chain",
+		Args:      true,
+		ArgsUsage: "<chain-id>",
```
Comment (bitwiseguy) [inline op-chain-ops/deployer/inspect/flags.go]: Might want to specify that you are looking for the l2 chain id here to avoid confusion

**Review by bitwiseguy — APPROVED** [review]
Overall looks good to me. Not a blocker for this pr, but will note that we are also maintaining an `AddressList` of all contracts within the `superchain-registry/superchain` module [here](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/superchain.go#L287). We should try to think of way to ensure those lists stay in sync, or a way to combine them and import the same struct where needed

File: op-chain-ops/deployer/inspect/flags.go (hunk 2 above)
Comment (mds1) [reply op-chain-ops/deployer/inspect/flags.go]: Good idea, fixed in ddd446fb4

**Comment by mds1** [comment]
> Overall looks good to me. Not a blocker for this pr, but will note that we are also maintaining an `AddressList` of all contracts within the `superchain-registry/superchain` module [here](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/superchain.go#L287). We should try to think of way to ensure those lists stay in sync, or a way to combine them and import the same struct where needed

Created an issue to track this here: https://github.com/ethereum-optimism/platforms-team/issues/343

**Pull request merged**